src/vanilla/internals.ts

```diff
@@ -490,8 +486,14 @@ const buildStore = (
         unmountCallbacks.size ||
         mountCallbacks.size
       )
-      if (hasError) {
-        throw error
+
+      if (errors.length === 1) {
+        throw errors[0]
+      } else if (errors.length > 1) {
+        throw new AggregateError(
+          errors,
+          'Multiple errors occurred during flushCallbacks',
+        )
```

Comment:
I would even go with this. Not sure about the drawback, but it's predictable.
```suggestion
      if (errors.length) {
        throw new AggregateError(errors)
```

Reply:
@dai-shi I think it's good way to go!
I'd change commit with it and force-push, thanks!